sqlmesh/core/console.py

```diff
@@ -493,14 +512,56 @@ def update_migration_progress(self, num_tasks: int) -> None:
         if self.migration_progress is not None and self.migration_task is not None:
             self.migration_progress.update(self.migration_task, refresh=True, advance=num_tasks)
 
-    def stop_migration_progress(self, success: bool = True) -> None:
+    def log_migration_status(self, success: bool = True) -> None:
+        """Log the migration status."""
+        if self.migration_progress is not None:
+            self.migration_progress = None
+            if success:
+                self.log_success("The migration has been completed successfully")
+
+    def stop_snapshot_migration_progress(self, success: bool = True) -> None:
         """Stop the migration progress."""
         self.migration_task = None
         if self.migration_progress is not None:
             self.migration_progress.stop()
-            self.migration_progress = None
             if success:
-                self.log_success("The migration has been completed successfully")
+                self.log_success("All snapshots have been migrated successfully")
+
+    def start_env_migration_progress(self, total_tasks: int) -> None:
+        """Indicates that a new environment migration has begun."""
+        if self.env_migration_progress is None:
+            self.env_migration_progress = Progress(
+                TextColumn("[bold blue]Migrating environments", justify="right"),
+                BarColumn(bar_width=40),
+                "[progress.percentage]{task.percentage:>3.1f}%",
+                "•",
+                srich.BatchColumn(),
+                "•",
+                TimeElapsedColumn(),
+                console=self.console,
```

Comment:
Nit: at this point I'd consider making a `Progress` factory function, which receives the progress' message as its argument, e.g. `Migrating environments`. It seems like the rest of the arguments are repeated and we have like ~4 places where we do this in console.py.

Reply:
Yes I considered that too, should I go ahead and refactor this?